servers/visual/shader_language.cpp

```diff
@@ -3875,8 +3875,8 @@ Error ShaderLanguage::_parse_shader(const Map<StringName, FunctionInfo> &p_funct
 					return ERR_PARSE_ERROR;
 				}
 
-				if (!uniform && (type < TYPE_FLOAT || type > TYPE_VEC4)) {
-					_set_error("Invalid type for varying, only float,vec2,vec3,vec4 allowed.");
+				if (!uniform && (type < TYPE_BOOL || type > TYPE_MAT4)) {
+					_set_error("Invalid type for varying, only float,vec2,vec3,vec4,mat2,mat3,mat4 allowed.");
```

Comment:
Also supporting `bvec2,bvec3,bvec4,int,ivec2,ivec3,ivec4` now with the `TYPE_BOOL` lower limit, they should be mentioned too.

At this point it might be better to rephrase though: "Invalid type for varying. The following types are supported: float, int, vec2, vec3, vec4, ivec2, ivec3, ivec4, bvec2, bvec3, bvec4, mat2, mat3, mat4."

Reply:
Ah.. its editing error

Reply:
I tried to enable ivec and other before TYPE_FLOAT but it must use flat keyword in order to compile, so I drop this idea

Reply:
Ok, feel free to merge it, I fixed it